### PR TITLE
Add support to upload a pkcs12 file

### PR DESCRIPTION
**Pull request by dasuni-30**
Empty File Uploader 
Screenshot 2024-01-24 at 15 13 42

Uploaded File
Screenshot 2024-01-24 at 15 14 29

### Related Issues
- https://github.com/wso2/product-is/issues/18993

### Related PRs
- None

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: .changeset/olive-bats-change.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+"@wso2is/react-components": patch
```
Comment (pavinduLakshan) [inline .changeset/olive-bats-change.md:2]: Let's bump the versions of the dependent packages as well.

File: apps/console/src/features/connections/components/edit/forms/components/common-pluggable-component-form.tsx [hunk 2]
```diff
@@ -60,6 +60,19 @@ export const CommonPluggableComponentForm: FunctionComponent<CommonPluggableComp
     // Used for field elements which needs to listen for any onChange events in the form.
     const [ dynamicValues, setDynamicValues ] = useState<CommonPluggableComponentInterface>(undefined);
     const [ customProperties, setCustomProperties ] = useState<string>(undefined);
+    const [ privateKeyValue, setPrivateKeyValue ] = useState<string>(undefined);
+
+    useEffect(() => {
+        dynamicValues?.properties?.map(
+            (prop: CommonPluggableComponentPropertyInterface) =>
+            {
+                if (prop.key === "google_prov_private_key") {
```
Comment (pavinduLakshan) [inline apps/console/src/features/connections/components/edit/forms/components/common-pluggable-component-form.tsx:69]: Shall we move "google_prov_private_key" to a constant [1]?

[1] https://github.com/wso2/identity-apps/blob/master/apps/console/src/features/connections/constants/connection-constants.ts

File: apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx [hunk 3]
```diff
@@ -493,9 +498,74 @@ export const getDropDownField = (eachProp: CommonPluggableComponentPropertyInter
         </>
     );
 };
+/* eslint-disable react-hooks/rules-of-hooks */
```
Comment (pavinduLakshan) [inline apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx:501]: Let's add a new line before the comment.

btw, curious to know the reason for disabling eslint hook rules here :)

```suggestion

/* eslint-disable react-hooks/rules-of-hooks */
```

File: apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx [hunk 4]
```diff
@@ -493,9 +498,74 @@ export const getDropDownField = (eachProp: CommonPluggableComponentPropertyInter
...
+export const getFilePicker = (eachProp: CommonPluggableComponentPropertyInterface,
+    propertyMetadata: CommonPluggableComponentMetaPropertyInterface,
+    testId?: string,
+    onCertificateChange?: ( newFile: string) => void): ReactElement => {
+
+    const [ data, setData ] = useState<string>(eachProp?.value);
+
+
```
Comment (pavinduLakshan) [inline apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx:509]:
```suggestion
```

File: apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx [hunk 5]
```diff
@@ -493,9 +498,74 @@ export const getDropDownField = (eachProp: CommonPluggableComponentPropertyInter
...
+            { (( data === "" || data?.length === 0 || eachProp?.value === undefined )) ? (
+                <>
+                    <Field
+                        name={ eachProp?.key || propertyMetadata?.key }
+                        label={ propertyMetadata?.displayName }
+                        required={ propertyMetadata?.isMandatory }
+                        requiredErrorMessage={ I18n.instance.t("console:develop.features.authenticationProvider." +
+                        "forms.common.requiredErrorMessage") }
+                        validation={ (value: string, validation: Validation) => {
+                            if (!FormValidation.url("https://www.sample.com?" + value)) {
```
Comment (pavinduLakshan) [inline apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx:521]: Is using `sample.com` thing here intentional?

File: apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx [hunk 6]
```diff
@@ -493,9 +498,74 @@ export const getDropDownField = (eachProp: CommonPluggableComponentPropertyInter
...
+                        } }
+                        type="filePicker"
+                        value={ data }
+                        key={ propertyMetadata?.key }
+                        disabled={ propertyMetadata?.isDisabled }
+                        readOnly={ propertyMetadata?.readOnly }
+                        data-testid={ `${ testId }-${ propertyMetadata?.key }` }
+                        fileStrategy={ P12_FILE_STRATEGY }
+                        uploadButtonText="Upload certificate File"
+                        dropzoneText="Drag and drop a certificate file here."
```
Comment (pavinduLakshan) [inline apps/console/src/features/connections/components/edit/forms/helpers/form-fields-helper.tsx:536]: Don't we need to i18n these strings?

File: apps/console/src/features/connections/components/edit/forms/helpers/pkcs12-file-field.tsx [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+} from "@wso2is/core/models";
+import { ResourceList, ResourceListActionInterface, ResourceListItem } from "@wso2is/react-components";
+import React, { FC, PropsWithChildren, ReactElement } from "react";
+import { Grid, Icon } from "semantic-ui-react";
+import {
+    CommonPluggableComponentMetaPropertyInterface,
+    CommonPluggableComponentPropertyInterface
+} from "../../../../models/connection";
+
+
```
Comment (pavinduLakshan) [inline apps/console/src/features/connections/components/edit/forms/helpers/pkcs12-file-field.tsx:30]:
```suggestion
```

File: apps/console/src/features/connections/components/edit/forms/helpers/pkcs12-file-field.tsx [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+/**
+ * Props interface of {@link IdpCertificatesList}
+ */
+export interface Pkcs12FileFieldProps extends IdentifiableComponentInterface {
+    /**
+     * Type of the template.
+     */
+    eachProp?: CommonPluggableComponentPropertyInterface;
+    propertyMetadata?: CommonPluggableComponentMetaPropertyInterface;
```
Comment (pavinduLakshan) [inline apps/console/src/features/connections/components/edit/forms/helpers/pkcs12-file-field.tsx:39]: Let's add doc comments for other props too.

File: apps/console/src/features/connections/components/edit/forms/helpers/pkcs12-file-field.tsx [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+     * @param index - Cert index.
+     */
+    const createCertificateActions = (index: number) => {
+        return [
+            {
+                "data-componentid": `${ testId }-delete-cert-${ index }-button`,
+                icon: "trash alternate",
+                onClick: () => handleDelete(),
+                popupText: "Delete",
+                type: "button"
```
Comment (pavinduLakshan) [inline apps/console/src/features/connections/components/edit/forms/helpers/pkcs12-file-field.tsx:71]: Shall we use i18n for these strings?

File: apps/console/src/features/connections/components/edit/forms/helpers/pkcs12-file-field.tsx [hunk 10]
```diff
@@ -0,0 +1,138 @@
...
+     * Handles the deletion of a certificate.
+     *
+     * @param certificateIndex - Index of the certificate to be deleted.
+     */
+    const handleDelete = () => {
+        onCertificateChange("");
+    };
+
+    return (
+        <>
```
Comment (pavinduLakshan) [inline apps/console/src/features/connections/components/edit/forms/helpers/pkcs12-file-field.tsx:97]: Unnecessary fragment

File: apps/console/src/features/connections/components/edit/forms/helpers/pkcs12-file-field.tsx [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+                                actionsFloated="right"
+                                avatar={ createCertificateResourceAvatar() }
+                                itemHeader={
+                                    "Private Key"
+                                }
+                                itemDescription={
+                                    "PKCS12 private key file"
+                                }
+                            />
+
```
Comment (pavinduLakshan) [inline apps/console/src/features/connections/components/edit/forms/helpers/pkcs12-file-field.tsx:124]:
```suggestion
```

**Comment by dasuni-30** [comment]
This PR was in the WIP state. i18n strings will be added in a follow-up PR.

**Review by JayaShakthi97 — APPROVED** [review]

**Pull request merged**